gci/componentmodel.py

```diff
@@ -65,7 +85,7 @@ class AccessType(enum.StrEnum):
 AccessTypeOrStr = typing.Union[AccessType, str]
 
 
-@dc(frozen=True, kw_only=True)
```

Comment:
I prefer to keep them frozen.
Most of the time mutating objects is a bad idea, and for the remaining cases where this might be valid, one can still use workarounds, e.g. `dataclasses.replace`

Reply:
I have to disagree. The way we use those model-classes, e.g. in CTT, we _do_ mutate. I do not actually see we gain benefits from keeping (parts of the model, anyways..) immutable.

I wanted to do some cleanups (like removal of as many as possible dependencies towards non-standard-libs), but also make the model-elements mutable, to thus improve ease-of use.